recipes-security/ssh-key-injection/ssh-key-injection.bb

```diff
@@ -0,0 +1,37 @@
+SUMMARY = "Recipe to inject SSH public keys"
+DESCRIPTION = "Injects specified SSH public keys into the image"
+LICENSE = "MIT"
+LIC_FILES_CHKSUM = "file://${COMMON_LICENSE_DIR}/MIT;md5=0835ade698e0bcf8506ecda2f7b4f302"
+
+RDEPENDS:${PN} = "dropbear"
+
+python () {
+    ssh_pubkey = d.getVar('SSH_PUBKEY')
+    
+    if ssh_pubkey is None:
+        origenv = d.getVar("BB_ORIGENV", False)
+        if origenv:
+            ssh_pubkey = origenv.getVar('SSH_PUBKEY')
+    
+    if ssh_pubkey:
+        d.setVar('SSH_PUBKEY', ssh_pubkey)
+        bb.note("SSH_PUBKEY is set to: %s" % ssh_pubkey)
+    else:
+        bb.fatal("SSH_PUBKEY must be set. Please provide an SSH public key.")
+}
```

Comment:
currently, building the image would crash if none was provided. We could change this to only a bb.warn if wanted and it would simply add 0 authorized ssh keys instead.
However, I would like later to have it as optional recipe itself such that we don't need to even add dropbear or ssh dependencies inside the final image at all. This way, we don't need to have the ssh attack surface anyway and the image would autonomously operate with no intervention as we envision it

Reply:
we should support building it without ssh key. keep in mind that this repo might get some public scrutiny, and people might be irritated by a requirement of adding SSH keys.

so yeah agree we should make it warn only now, or just info even, and later on have a build without dropbear (that can wait a little while).